android/runtime/v8/src/native/modules/APIModule.cpp

```diff
@@ -115,7 +115,19 @@ void APIModule::logInfo(const FunctionCallbackInfo<Value>& args)
 	Isolate* isolate = args.GetIsolate();
 	HandleScope scope(isolate);
 	v8::String::Utf8Value message(isolate, APIModule::combineLogMessages(args));
-	APIModule::logInternal(LOG_LEVEL_INFO, LCAT, *message);
+
+	std::string cppStr(*message);
+
+	int maxChunk = 4050;
+	int chunkCount = cppStr.length() / maxChunk;
+	for (int i = 0; i <= chunkCount; i++) {
+		int max = maxChunk * (i + 1);
+		if (max >= cppStr.length()) {
+			APIModule::logInternal(LOG_LEVEL_INFO, LCAT, cppStr.substr(maxChunk * i).c_str());
+		} else {
+			APIModule::logInternal(LOG_LEVEL_INFO, LCAT, cppStr.substr(maxChunk * i , max).c_str());
```

Comment:
The second arg, `max` is should be the number of characters to include in the substring. Each iteration, `max` is growing and growing. I think you would be fine omitting the `if` and always passing in `maxChunk` for the second arg.

Reply:
> I think you would be fine omitting the if and always passing in maxChunk for the second arg.

wouldn't that crash if the len is < maxChunk? Have to recheck. If that is not the case then yes, I can always use the 2nd `if` part

Reply:
you are right, not needed. Works fine with short strings:

> Number of characters to include in the substring **(if the string is shorter, as many characters as possible are used**).